File: samples/python/primality/prime.py
import math
import sys


def is_prime(number):
    limit = math.sqrt(number)
    start = 2
    while start <= limit:
        if number % start == 0:
            return False
        start += 1
    return True


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Error: Usage python prime.py <number>")
        sys.exit(1)
    try:
        number = float(sys.argv[1])
    except ValueError:
        print("Error: Expected a number. Received", sys.argv[1])
        sys.exit(2)
    print(is_prime(number))

File: samples/kotlin/04prime/prime.kt
import kotlin.math.sqrt


fun isPrime(num: Int): Boolean {
    val limit = sqrt(num.toDouble()).toInt();
    for (n in 2..limit) {
        if (num.rem(n) == 0) {
            return false;
        }
    }
    return true;
}


fun main(args: Array<String>) {
    if (args.size == 0) {
        System.err.println("Error: Usage `kotlin sqrt <number> [<number> ...]`");
        System.exit(1);
    }
    for (arg in args) {
        var number: Int = 0;
        try {
            number = arg.toInt();
        } catch (e: NumberFormatException) {
            System.err.println(String.format("Error: Expected to receive a number. Received '%s'.", arg));
            System.exit(1);
        }
        println(String.format("%d => %s", number, if (isPrime(number)) "true" else "false"));
    }
}
